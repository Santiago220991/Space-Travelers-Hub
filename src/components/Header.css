.header {
  padding-left: 2%;
  padding-right: 2%;
  border-bottom: solid 1px rgb(184, 178, 178);
  margin-bottom: 20px;
}

.active {
  color: rgb(89, 0, 255);
  text-decoration: underline;
}

.none {
  color: rgb(0, 195, 255);
  text-decoration: none;
}

.nav {
  display: flex;
  align-items: center;
  justify-content: flex-start;
}

.logo-title {
  display: flex;
  align-items: center;
  font-size: 32px;
  font-weight: bolder;
  column-gap: 20px;
}

.menu {
  display: flex;
  column-gap: 30px;
  margin-left: calc(50vw - 200px);
}
