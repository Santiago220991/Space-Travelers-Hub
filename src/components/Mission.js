import PropTypes from 'prop-types';
import { useDispatch } from 'react-redux';
import { ToggleMissionAction } from '../redux/missions/missions';

function Mission(props) {
  const dispatch = useDispatch();
  function ToggleMission(event) {
    const { id } = event.target.parentNode.parentNode;
    dispatch(ToggleMissionAction(id));
  }
  const {
    id, name, description,
  } = props;
  return (
    <div className="row" id={id}>
      <div className="column1 MissionName cell">{name}</div>
      <div className="column2 MissionDesc cell">{description}</div>
      <div className="column3 cell"><span className="ActiveMember">Active Member</span></div>
      <div className="column4 cell">
        <button type="button" className="LeaveMission" onClick={ToggleMission}>
          Leave Mission
        </button>
      </div>
    </div>
  );
}

export default Mission;

Mission.propTypes = { id: PropTypes.string.isRequired };
Mission.propTypes = { name: PropTypes.string.isRequired };
Mission.propTypes = { description: PropTypes.string.isRequired };
