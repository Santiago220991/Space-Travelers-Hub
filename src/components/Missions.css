.missions {
  display: flex;
  flex-direction: column;
}

.titles {
  display: flex; 
}

.row {
  display: flex; 
}

.MissionName {
  padding: 10px;
  width: 10%;
}

.MissionDesc {
  padding: 10px;
  width: 60%;
}

.MissionStatus {
  padding: 10px;
  width: 15%;
}

.MissionButton {
  padding: 10px;
  width: 15%;
}
