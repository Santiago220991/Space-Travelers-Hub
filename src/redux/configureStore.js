import { configureStore } from '@reduxjs/toolkit';
import thunk from 'redux-thunk';
import RocketReducer from './rockets/rockets';
import MissionsReducer from './missions/missions';

const Store = configureStore({
  reducer: {
    rocket: RocketReducer,
    missions: MissionsReducer,
  },
  middleware: [thunk],
});

export default Store;
