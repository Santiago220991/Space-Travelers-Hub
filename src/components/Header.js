import { NavLink } from 'react-router-dom';
import planetLogo from '../planet.png';
import './Header.css';

function Header() {
  return (
    <header className="header">
      <div>
        <img src={planetLogo} width="20" height="20" alt="" />
      </div>
      <nav className="nav">
        <div className="Rockets">
          <NavLink className={(navData) => (navData.isActive ? 'active' : 'none')} to="/">Rockets</NavLink>
        </div>
        <div className="Missions">
          Missions
        </div>
        <div className="Profile">
          <NavLink className={(navData) => (navData.isActive ? 'active' : 'none')} to="/Profile">Profile</NavLink>
        </div>
      </nav>
    </header>
  );
}

export default Header;
