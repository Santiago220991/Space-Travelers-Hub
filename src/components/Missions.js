import './Missions.css';

const Missions = () => (
  <div className="missions">
    <div className="titles">
      <div className="MissionName">Mission</div>
      <div className="MissionDesc">Description</div>
      <div className="MissionStatus">Status</div>
      <div className="MissionButton">&nbsp;</div>
    </div>
    <div className="row">
      <div className="MissionName">Thaicom</div>
      <div className="MissionDesc">
        Lorem Ipsum is simply dummy text of the printing and typesetting industry.
        Lorem Ipsum has been the industry&apos;s standard dummy text ever since the 1500s,
        when an unknown printer took a galley of type and scrambled it to make a type specimen book.
        It has survived not only five centuries, but also the leap into electronic typesetting,
        remaining essentially unchanged. It was popularised in the 1960s with the release of
        Letraset sheets containing Lorem Ipsum passages, and more recently with desktop publishing
        software like Aldus PageMaker including versions of Lorem
      </div>
      <div className="MissionStatus ActiveMember">Active Member</div>
      <div className="MissionButton"><button type="button" className="LeaveMission">Leave Mission</button></div>
    </div>
  </div>
);

export default Missions;
