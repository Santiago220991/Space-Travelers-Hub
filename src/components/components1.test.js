import React from 'react';
import { Provider } from 'react-redux';
import { MemoryRouter } from 'react-router-dom';
import renderer from 'react-test-renderer';
import configureStore from '../redux/configureStore';
import Header from './Header';
import Missions from './Missions';
import Mission from './Mission';

test('Header renders correctly', () => {
  const tree = renderer
    .create(
      <MemoryRouter>
        <Header />
      </MemoryRouter>,
    )
    .toJSON();

  expect(tree).toMatchSnapshot();
});

test('Missions renders correctly', () => {
  const tree = renderer
    .create(<Provider store={configureStore}><Missions /></Provider>)
    .toJSON();

  expect(tree).toMatchSnapshot();
});

test('Mission renders correctly', () => {
  const tree = renderer
    .create(
      <Provider store={configureStore}>
        <Mission key="FF3864" id="FF3864" name="Mission1" description="Description1" member={false} />
      </Provider>,
    )
    .toJSON();

  expect(tree).toMatchSnapshot();
});
