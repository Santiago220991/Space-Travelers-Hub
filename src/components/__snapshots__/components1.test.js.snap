// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Header renders correctly 1`] = `
<header
  className="header"
>
  <nav
    className="nav"
  >
    <div
      className="logo-title"
    >
      <img
        alt=""
        height="40"
        src="planet.png"
        width="40"
      />
      <p>
        Space Travelers' Hub
      </p>
    </div>
    <div
      className="menu"
    >
      <div
        className="Rockets"
      >
        <a
          aria-current="page"
          className="active"
          href="/"
          onClick={[Function]}
        >
          Rockets
        </a>
      </div>
      <div
        className="Missions"
      >
        <a
          className="none"
          href="/Missions"
          onClick={[Function]}
        >
          Missions
        </a>
      </div>
      <div
        className="Profile"
      >
        <a
          className="none"
          href="/Profile"
          onClick={[Function]}
        >
          Profile
        </a>
      </div>
    </div>
  </nav>
</header>
`;

exports[`Mission renders correctly 1`] = `
<div
  className="row"
  id="FF3864"
>
  <div
    className="column1 MissionName cell"
  >
    Mission1
  </div>
  <div
    className="column2 MissionDesc cell"
  >
    Description1
  </div>
  <div
    className="column3 cell"
  >
    <span
      className="NotAMember"
    >
      NOT A MEMBER
    </span>
  </div>
  <div
    className="column4 cell"
  >
    <button
      className="JoinMission"
      onClick={[Function]}
      type="button"
    >
      Join Mission
    </button>
  </div>
</div>
`;

exports[`Missions renders correctly 1`] = `
<div
  className="missions"
>
  <div
    className="titles"
  >
    <div
      className="column1 cell"
    >
      Mission
    </div>
    <div
      className="column2 cell"
    >
      Description
    </div>
    <div
      className="column3 cell"
    >
      Status
    </div>
    <div
      className="column4 cell"
    >
      Â 
    </div>
  </div>
</div>
`;
